### PR TITLE
refactor: search bar component

### LINKED ISSUE #3883: Refactor: SearchBar - splitting into smaller components
Currently, our search SearchBar.vue is at 909 lines of code (bad, hard to debug etc.)

We'd like to see this split up into smaller components, to give an easily demonstrated example, there are at least three different logical chunks I could see being components:
image

Somebody with previous experience in similar refactor efforts could pick this up, go over the code and find more elegant solution to our current SearchBar.vue implementation.

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring

### What's new?

- [x] PR closes #3883
- [ ] Requires deployment <rubick/magick_issue>
- [ ] <brief_description_of_what_I've_added>

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

### Optional

- [ ] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [ ] I've tested PR on mobile and everything seems works
- [ ] I found edge cases
- [ ] I've written some unit tests 🧪

### Had issue bounty label?

- [X] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)

### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### CODE REVIEW

**Comment by roiLeo** [comment]
Can we have a component with only `SearchBar` input? without switch, filters & range

**Comment by Jarsen136** [comment]
> Can we have a component with only `SearchBar` input? without switch, filters & range

Yes, that's a good idea. 
Already updated.

**Comment by petersopko** [comment]
@Jarsen136 it's not a range slider anymore, some of the functions/variables may have been left there, but maybe let's call it something more appropriate?

**Comment by Jarsen136** [comment]
> @Jarsen136 it's not a range slider anymore, some of the functions/variables may have been left there, but maybe let's call it something more appropriate?

Yup, will check it

File: components/search/SearchSuggestion.vue [hunk 1]
```diff
@@ -0,0 +1,498 @@
...
+    cursor: pointer;
+  }
+
+  .media-content {
+    .name {
+      max-width: 34ch;
+      overflow: hidden;
+      text-overflow: ellipsis;
+    }
+  }
```
Comment (prachi00) [inline components/search/SearchSuggestion.vue:496]: This file still has 200+ lines of code, would it be possible to refactor it further?

Comment (Jarsen136) [reply components/search/SearchSuggestion.vue:488]: Updated. Though there is no big stuff I could refactor in this file.

**Comment by petersopko** [comment]
I'd change the names of two files:
SearchBar.vue -> Search.vue
SearchBarInput.vue -> SearchBar.vue
ref:
image

**Comment by Jarsen136** [comment]
> I'd change the names of two files: SearchBar.vue -> Search.vue SearchBarInput.vue -> SearchBar.vue ref: ![image]

Done!

**Review by prachi00 — APPROVED** [review]

**Comment by petersopko** [comment]
@Jarsen136 the last thing here is, let's try to fix the tests for Buy Now button, I assume something was changed in this regard and that's why the tests are failing right now.

**Comment by petersopko** [comment]
@vikiival this lgtm, can you take a look ? @prachi00 said it's good, but I'm not sure about the length of the file (or whether there's a possibility for shortening it further)

File: components/search/Search.vue [hunk 2]
```diff
@@ -0,0 +1,365 @@
+<template>
```
Comment (vikiival) [inline components/search/Search.vue:1]: Why does this have like 400 LOC?

File: components/search/SearchCollection.vue [hunk 3]
```diff
@@ -58,7 +58,7 @@ import KeyboardEventsMixin from '~/utils/mixins/keyboardEventsMixin'
     BasicSwitch: () => import('@/components/shared/form/BasicSwitch.vue'),
   },
 })
-export default class CollectionSearchBar extends mixins(KeyboardEventsMixin) {
+export default class extends mixins(KeyboardEventsMixin) {
```
Comment (vikiival) [inline components/search/SearchCollection.vue:61]: You miss class name here. It will emit error in Vue

**Review by vikiival — CHANGES_REQUESTED** [review]
Please do more proper code split in `components/search/Search.vue`

It still contains like 400 LOC

**Comment by vikiival** [comment]
>  but I'm not sure about the length of the file

Yeah it's too long

File: components/search/SearchCollection.vue (hunk 3 above)
Comment (Jarsen136) [reply components/search/SearchCollection.vue]: Added

**Comment by Jarsen136** [comment]
> Please do more proper code split in `components/search/Search.vue`
> 
> It still contains like 400 LOC

I have tried to reduce it to about 300 LOC : )

**Comment by yangwao** [comment]
Why it's showing the third collection when there is no "Moon" in the name of the collection?

image

**Comment by Jarsen136** [comment]
> Why it's showing the third collection when there is no "Moon" in the name of the collection?
> 
> image

For the third collection https://deploy-preview-3894--koda-nuxt.netlify.app/movr/collection/0x2f26efdb7233a014715ce6e895aa67d846d93f1e.

image

Its original name is `MoonShroomiz ` so that it's showing in the search result.
After fetching the metadata of the this collection https://kodadot.mypinata.cloud/ipfs/QmXvfjHAiaJHTu5692YofHZunvRJdZiPpcepw8pu4B3wuf/1.json.

You could see that its final name is `Patient Naoko`.
image


IMO it's a bug related to the subsquid 002-rc0 side. @vikiival Could you pls take a look 👀

**Comment by vikiival** [comment]
> IMO it's a bug related to the subsquid 002-rc0 side.

https://github.com/kodadot/click/issues/34

**Comment by yangwao** [comment]
Merging in prior to continue to bring search bar on the landing page - #3844

**Comment by yangwao** [comment]
pay 100

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $100 @ 51.04 USD/KSM ~ 1.959 $KSM
🧗 [Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG](https://kusama.subscan.io/account/Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)
🔗 [0xd3c42832c1ca83014101948aeb017d49d60853ec60b8d6d01c9559b38c5d04e4](https://kusama.subscan.io/extrinsic/0xd3c42832c1ca83014101948aeb017d49d60853ec60b8d6d01c9559b38c5d04e4)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**